test/functional/write_concern.test.js

```diff
@@ -117,4 +117,56 @@ describe('Write Concern', function () {
         return client.close();
       });
   });
+
+  describe('details property', () => {
+    let client;
+
+    beforeEach(async function () {
+      client = this.configuration.newClient({ monitorCommands: true });
+      await client.connect();
+    });
+
+    afterEach(async () => {
+      if (client) await client.close();
+    });
+
+    it('should provide details on why document validation failed', {
+      metadata: { requires: { mongodb: '>=5.0.0' } },
+      async test() {
+        try {
+          await client.db().collection('wc_details').drop();
+        } catch {
+          // don't care
+        }
+
+        const collection = await client
+          .db()
+          .createCollection('wc_details', { validator: { x: { $type: 'string' } } });
+
+        const evCapture = require('events').once(client, 'commandSucceeded');
```

Comment:
since node will cache any requires, I don't think hiding a require inside a test instead of declaring it up top is a good idea; we should also be careful to reset event listeners in an after block in case it isn't triggered in the test